app/views/recipes/_form.html.erb

```diff
@@ -13,33 +10,47 @@
         </div>
       </div>
 
-      <div class="relative">
-        <div class="absolute inset-0 flex items-center" aria-hidden="true">
-          <div class="w-full border-t border-gray-300"></div>
-        </div>
-        <div class="relative flex justify-start">
-          <span class="bg-white pr-2 text-sm text-gray-500">Ingredients</span>
-        </div>
-      </div>
+      <div data-controller="template-manager">
 
-      <div id="ingredients" class="space-y-3" data-streams-target="container">
-        <% recipe.ingredients.each do |ingredient| %>
-          <%= render "recipes/ingredient_fields", ingredient: ingredient %>
-        <% end %>
-      </div>
+        <!-- Ingredients section-->
+        <div class="relative">
+          <div class="absolute inset-0 flex items-center" aria-hidden="true">
+            <div class="w-full border-t border-gray-300"></div>
+          </div>
+          <div class="relative flex justify-start">
+            <span class="bg-white pr-2 text-sm text-gray-500">Ingredients</span>
+          </div>
+        </div>
+        <!-- Ingredients template-->
+        <template data-template-manager-target="template">
+          <%= form.fields_for "ingredients_attributes", index: "replace_with_epoch" do |template_form| %>
```

Comment:
the "ingredients_attributes" string comes from the name that rails will give to fields within the  `form.fields_for ingredients` block. I wish there was a more conventional way to do it but I couldn't find it.

Reply:
Conventional as in, I don't like that it's hardcoded.